docs/lib-documentation.md

```diff
@@ -58,39 +58,30 @@ Gets the [LambdaContext][lambdacontext] associated with an already-running Lambd
 
 * `lambdaContext`: The [LambdaContext][lambdacontext] representation of the running Lambda container, which allows execution of the Lambda function.
 
-### registerStaticFiles(Map\<StaticFileID, String\> staticFileFilepaths)
-Registers the given files with the Kurtosis engine, so they can be loaded inside a service's filespace via [ContainerCreationConfig.usedStaticFiles][containercreationconfig_usedstaticfiles] and [ServiceContext.loadStaticFiles][servicecontext_loadstaticfiles].
-
-**Args**
-
-* `staticFileFilepaths`: A map of static_file_id -> absolute_filepath containing the files to register and the ID by which they'll be referenced via [ContainerCreationConfig.usedStaticFiles][containercreationconfig_usedstaticfiles] and [ServiceContext.loadStaticFiles][servicecontext_loadstaticfiles].
-
 ### registerFilesArtifacts(Map\<FilesArtifactID, String\> filesArtifactUrls)
-Downloads the given files artifacts to the Kurtosis engine, associating them with the given IDs, so they can be mounted inside a service's filespace at creation time via [ContainerCreationConfig.filesArtifactMountpoints][containercreationconfig_filesartifactmountpoints].
+Downloads the given files artifacts to the Kurtosis engine, associating them with the given IDs, so they can be mounted inside a service's filespace at creation time via [ContainerConfig.filesArtifactMountpoints][containerconfig_filesartifactmountpoints].
 
 **Args**
 
-* `filesArtifactUrls`: A map of files_artifact_id -> url, where the ID is how the artifact will be referenced in [ContainerCreationConfig.filesArtifactMountpoints][containercreationconfig_filesartifactmountpoints] and the URL is the URL on the web where the files artifact should be downloaded from.
+* `filesArtifactUrls`: A map of files_artifact_id -> url, where the ID is how the artifact will be referenced in [ContainerConfig.filesArtifactMountpoints][containerconfig_filesartifactmountpoints] and the URL is the URL on the web where the files artifact should be downloaded from.
 
-### addServiceToPartition(ServiceID serviceId, PartitionID partitionId, [ContainerCreationConfig][containercreationconfig] containerCreationConfig, Func(String ipAddr, Map\<String, String\> generatedFileFilepaths, Map\<String, String\> staticFileFilepaths) -\> [ContainerRunConfig][containerrunconfig] generateRunConfigFunc) -\> ([ServiceContext][servicecontext] serviceContext, Map\<String, PortBinding\> hostPortBindings)
-Starts a new service in the network with the given service ID, inside the partition with the given ID, using the given config factory.
+### addServiceToPartition(ServiceID serviceId, PartitionID partitionId, Func(String ipAddr, SharedPath sharedDirectory) -\> [ContainerConfig][containerconfig] containerConfigSupplier) -\> ([ServiceContext][servicecontext] serviceContext, Map\<String, PortBinding\> hostPortBindings)
```

Comment:
make sure to make `SharedPath ` a link to the SharedPath docs (the only ones I don't link are simple types, like `String` or `ServiceID` (which is basically just a string)